lib/ts/recipe/session/sessionAuth.tsx

```diff
@@ -47,125 +49,96 @@ const SessionAuth: React.FC<Props> = ({ children, ...props }) => {
 
     const session = useRef(Session.getInstanceOrThrow());
 
-    const setInitialContext = useCallback(async (): Promise<void> => {
-        const sessionExists = await doesSessionExist();
+    // on mount
+    useEffect(() => {
+        let cancelUseEffect = false;
+        async function setInitialContextAndMaybeRedirect() {
```

Comment:
I'm doing a slight refactor of this function to avoid so many `if..else` nesting levels and separate concerns better.

Reply:
I think this function is fine the way it is. It calculates the context initially and acts based on what the value is. Please do not change it unless you see a bug in there.. no point in doing this type of refactor. Really no point.. Having nested `if..else` is 100% fine.